setup.py

```diff
@@ -69,7 +71,8 @@ def finalize_options(self):
         pass
 
     def run(self):
-        os.system('rm -vrf ./build ./dist ./*.pyc ./*.tgz ./*.egg-info')
+        os.system('./docker_build.sh')
```

Comment:
Not sure how I feel about `python setup.py test` invoking docker.

Reply:
It's a big TODO here (above). It's not how it will look like eventually. I have not yet looked at it but I am quite sure invoking docker in setup.py is a bad idea.

Reply:
I will keep it open. I am not sure what you think about it @fokko - but I think we have to limit the tests run via `python setup.py test` to only "real" unit tests - i.e. those that do not require external dependencies. I am not sure if we want to do it now or rather as separate PR though. Or maybe we should run the whole suite of tests (in which case using Docker compose for that might seem like a good idea). 

I would like to understand how `python setup.py test` is actually used now? Is it part of the release process to run it and prerequisite for release to have it green? Or maybe it's used in some other places? I haven't used it at all so far (that's why I kind of neglected it so far). Any hints/suggestions here?